doc/multipage-and-animated-images.md

```diff
@@ -0,0 +1,216 @@
+```
+
+Points to note:
+
+- By default, libvips will just read the first page from an animated or
+  multipage image. You pass `[n=-1]` to the loader to get all pages (or
+  frames) in the animation. You can pick out a single page or range of
+  pages with perhaps `[page=4]` and `[page=2,n=2]`.
+
+- `page-height` is the vertical size of each frame within the overall image
+  (2560 pixels high in this case).
+
+- `n-pages` is the number of pages (or frames) in this animation. Obviously
+  `n-pages * frame-height == height`, or in this case 320 * 8 == 2560.
+
+- `loop` is the number of times the animation should loop before stopping.
+  Zero means "never stop looping".
+
+- `delay` is an optional array with a time in milliseconds which each frame
+  should display for.
+
+You'll see a similar set of metadata for a multipage image, such as a PDF:
+
+```
+$ vipsheader -a nipguide.pdf[n=-1]
+nipguide.pdf: 595x48836 uchar, 4 bands, srgb, pdfload
+width: 595
+height: 48836
+bands: 4
+format: uchar
+coding: none
+interpretation: srgb
+xoffset: 0
+yoffset: 0
+xres: 2.83465
+yres: 2.83465
+filename: nipguide.pdf
+vips-loader: pdfload
+page-height: 842
+pdf-n_pages: 58
+n-pages: 58
+pdf-creator: TeX
+pdf-producer: pdfTeX-1.40.16
+```
+
+Now there's no `loop` or `delay` since this is not animated, but `n-pages` and
+`page-height` are set. In just the same way, you can load all pages, a single
+page or a range of pages.
+
+This all assumes that every page (or frame) has the same dimensions. If
+they don't (this can commonly happen with PDF and TIFF), you have to read
+pages one by one.
+
+# Writing multipage images
+
+As long as these various pieces of metadata are set, you can write animated
+and multipage images in the obvious way. For example:
+
+```
+$ vips copy nipguide.pdf[n=-1] x.gif
+```
+
+This will take the 58-page PDF and render a 58-frame animation.  This only
+works because this specific PDF has pages which are all the same size --
+PDFs with (for example) a mix of portrait and landscape pages can't be
+handled like this.
+
+More usefully, you could convert a GIF to WebP with:
+
+```
+$ vips copy silver-gear-cogs-animation-5.gif[n=-1] silver.webp
+```
+
+To write an animated or multipage image programmatically, you need to
+construct the tall, thin image and set the metadata. For example:
+
+```
+$ vips arrayjoin "k2.jpg k4a.png" x.tif[page-height=2048] --across=1
+```
+
+Provided that the images are both 2048 pixels high, this will write a
+two-page TIFF.
+
+In Python you could write something like:
+
+```python
+#!/usr/bin/env python3
+
+import sys
+import pyvips
+
+# the input images -- assume these are all the same size
+images = [pyvips.Image.new_from_file(filename, access="sequential")
+          for filename in sys.argv[2:]]
+
+# frame delays are in milliseconds
+delay_array = [300] * len(images)
+
+animation = pyvips.Image.arrayjoin(images, across=1)
+animation.set_type(pyvips.GValue.gint_type, "page-height", images[0].height)
```

Comment:
It might be worth including the `loop` and `n-pages` metadata here as well.
```suggestion
animation.set_type(pyvips.GValue.gint_type, "loop", 10)
animation.set_type(pyvips.GValue.gint_type, "n-pages", len(images))
animation.set_type(pyvips.GValue.gint_type, "page-height", images[0].height)
```